### PR TITLE
Detail plan for replacing Travis with GH Actions

### PR DESCRIPTION
**Pull request by mbestavros**
Mapped issue #18.

### CODE REVIEW

**Comment by lukehinds** [comment]
awesome, thanks @mbestavros 

@mpeters I reckon we can move this to in-progress and Mark cracks on with the work. Its given we need to move away from travis.

**Review by lukehinds — APPROVED** [review]

**Comment by mpeters** [comment]
> awesome, thanks @mbestavros
> 
> @mpeters I reckon we can move this to in-progress and Mark cracks on with the work. Its given we need to move away from travis.

Agreed

File: 18_migrate_ci_to_gh_actions.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+to Actions entirely.
+
+In particular, Github Actions is:
+
+- Directly integrated into Github's UI
+- Fast -- jobs start quickly, and most lead time happens at compilation or
+  environment setup
+- Extensible -- there are thousands of actions available for a wide variety of
+  tasks
+- Fits the fork-request-merge model in the way developers expect modern CI to
```
Comment (mpeters) [inline 18_migrate_ci_to_gh_actions.md:100]: Ending with a preposition makes it seem like it's missing a word at the end of this sentence.

File: 18_migrate_ci_to_gh_actions.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+enhancement ecosystem.
+
+How will security be reviewed and by whom?
+-->
+
```
Comment (mpeters) [inline 18_migrate_ci_to_gh_actions.md:156]: Maybe at least a "TBD" comment or something for these blank sections?

File: 18_migrate_ci_to_gh_actions.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+### Upgrade / Downgrade Strategy
+
+Github Actions configs can be updated by simply changing their workflow file in
```
Comment (mpeters) [inline 18_migrate_ci_to_gh_actions.md:185]: I think this section is supposed to be more about how the upgrade/downgrade strategy for the enhancement will work. For instance, how is the "switch" pulled to use GitHub Actions and not Travis done? How do we switch back if it didn't work out?

File: 18_migrate_ci_to_gh_actions.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+tests.
+
+## Drawbacks
+
+<!--
+Why should this enhancement _not_ be implemented?
+-->
+
+## Alternatives
+
```
Comment (mpeters) [inline 18_migrate_ci_to_gh_actions.md:199]: You mentioned above some alternatives to Travis but didn't list them. Not too important, but maybe a short list here would work.

**Review by mpeters — APPROVED** [review]
None of my requested changes are show stoppers and we can definitely consider this a "living" document

**Comment by mpeters** [comment]
Still lgtm :)

**Pull request merged**